File: FinalBootMVC/src/main/java/com/springboot/pkg/model/Formation.java
package com.springboot.pkg.model;

import java.util.Date;
import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonView;
import com.springboot.pkg.model.jsonview.JsonViews;


@Entity
public class Formation {

	@Id
	@SequenceGenerator(name = "formationSeq", sequenceName = "formation_seq", allocationSize = 1)
	@GeneratedValue(generator = "formationSeq", strategy = GenerationType.SEQUENCE)
	@JsonView(JsonViews.Common.class)
	private Long id;
	
	@Temporal(TemporalType.DATE)
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@JsonView(JsonViews.Common.class)
	private Date dateDebut;
	
	@Temporal(TemporalType.DATE)
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@JsonView(JsonViews.Common.class)
	private Date dateFin;


	@OneToMany (mappedBy = "formation")	
	private Set<Module> modules;

	@OneToMany(mappedBy="formation")
	private Set<Stagiaire> stagiaires;

	public Formation() {

	}

	public Long getId() { 
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Date getDateDebut() {
		return dateDebut;
	}

	public void setDateDebut(Date dateDebut) {
		this.dateDebut = dateDebut;
	}

	public Date getDateFin() {
		return dateFin;
	}

	public void setDateFin(Date dateFin) {
		this.dateFin = dateFin;
	}


	public Set<Stagiaire> getStagiaires() {
		return stagiaires;
	}

	public void setStagiaires(Set<Stagiaire> stagiaires) {
		this.stagiaires = stagiaires;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Formation other = (Formation) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

	 

}

File: FinalAngular/src/app/model/formation.ts
import { Module } from './module';
import { Stagiaire } from './stagiaire';
export class Formation {

    constructor(private _id?: number, private _dateDebut?: Date, private _dateFin?: Date,
        private _modules?: Array<Module>, private _stagiaires?: Array<Stagiaire>) {

    }

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get dateDebut(): Date {
        return this._dateDebut;
    }

    public set dateDebut(value: Date) {
        this._dateDebut = value;
    }

    public get dateFin(): Date {
        return this._dateDebut;
    }

    public set dateFin(value: Date) {
        this._dateDebut = value;
    }

    public get modules(): Array<Module> {
        return this._modules;
    }

    public set modules(value: Array<Module>) {
        this._modules = value;
    }

    public get stagiaires(): Array<Stagiaire> {
        return this._stagiaires;
    }

    public set stagiaires(value: Array<Stagiaire>) {
        this._stagiaires = value;
    }
}
